language/move-prover/interpreter/src/concrete/player.rs

```diff
@@ -0,0 +1,1357 @@
+    let local_state = exec_function(&ctxt, typed_args, global_state)?;
+    let termination = local_state.into_termination_status();
+    let return_vals = match termination {
+        TerminationStatus::Abort(abort_info) => {
+            return Err(abort_info.into_err());
+        }
+        TerminationStatus::Return(return_vals) => return_vals,
+        TerminationStatus::None | TerminationStatus::PostAbort(_) => unreachable!(),
+    };
+    Ok(return_vals)
+}
+
+/// Execute a function with the type arguments and value arguments.
+fn exec_function(
+    ctxt: &FunctionContext,
+    typed_args: Vec<RefTypedValue>,
+    global_state: &mut GlobalState,
+) -> PartialVMResult<LocalState> {
+    let target = &ctxt.target;
+    let env = target.global_env();
+
+    // discover and validate local slots
+    let param_decls = target.func_env.get_parameters();
+    if cfg!(debug_assertions) {
+        assert_eq!(param_decls.len(), typed_args.len());
+        assert!(param_decls.len() <= target.get_local_count());
+    }
+
+    let mut local_slots = vec![];
+    for (i, typed_arg) in typed_args.into_iter().enumerate() {
+        let name = env
+            .symbol_pool()
+            .string(target.get_local_name(i))
+            .to_string();
+
+        // check that types for local slots is compatible with the declared parameter type
+        if cfg!(debug_assertions) {
+            let local_ty = target.get_local_type(i);
+            let param_decl_ty = &param_decls.get(i).unwrap().1;
+            if local_ty != param_decl_ty {
+                assert!(matches!(
+                            param_decl_ty,
+                            MT::Type::Reference(false, base_ty)
+                            if local_ty == base_ty.as_ref()));
+            }
+            let ty = convert_model_local_type(env, local_ty, &ctxt.ty_args);
+            assert_eq!(&ty, typed_arg.get_ty());
+        }
+
+        let slot = LocalSlot::new_arg(name, typed_arg.deref());
+        local_slots.push(slot);
+    }
+    for i in param_decls.len()..target.get_local_count() {
+        let name = env
+            .symbol_pool()
+            .string(target.get_local_name(i))
+            .to_string();
+        let ty = convert_model_local_type(env, target.get_local_type(i), &ctxt.ty_args);
+        let slot = LocalSlot::new_tmp(name, ty);
+        local_slots.push(slot);
+    }
+
+    // execute the bytecode stepwise
+    let instructions = target.get_bytecode();
+    let mut local_state = LocalState::new(local_slots);
+    while !local_state.is_terminated() {
+        let pc = local_state.get_pc() as usize;
+        let bytecode = instructions.get(pc).unwrap();
+        exec_bytecode(ctxt, bytecode, &mut local_state, global_state)?;
+    }
+    Ok(local_state)
+}
+
+fn exec_bytecode(
+    ctxt: &FunctionContext,
+    bytecode: &Bytecode,
+    local_state: &mut LocalState,
+    global_state: &mut GlobalState,
+) -> PartialVMResult<()> {
+    match bytecode {
```

Comment:
nit: perhaps `use Bytecode::*` for brevity

Reply:
I am not a big fan of the `::*` notion as well as placing imports inside the function. But I do see the brevity, so, it is a hard choice.